torch/csrc/jit/serialization/source_range_serialization.cpp

```diff
@@ -118,9 +117,18 @@ void ConcreteSourceRangeUnpickler::unpickle() {
   unpickled_records = std::make_shared<SourceRangeRecords>();
   for (auto& val : ivalues) {
     auto tup_elems = val.toTuple()->elements();
-    int64_t offset = tup_elems[0].toInt();
-    auto source_range = deserializer->deserialize(tup_elems[1]);
-    unpickled_records->emplace_back(offset, std::move(source_range));
+    if (tup_elems.size() == 2) {
+      // This path is for BC.
+      // After this change debug_pkl stores {byte_offset, source_range_tag,
```

Comment:
curious: does the order of `source_range_tag` and `source_range` matter here? do we intentionally swap the order to highlight the format difference? if we always ignore `source_range_tag` on the read path here, isn't it easier to simply append `source_range_tag` after `source_range`?

Reply:
Thats a good point. There is not particular reason as to why tag is stored first. We can just append it. But is there a reason for one vs. the other besides few less lines of code change? Happy to make the change though.

Reply:
Just in terms of entropy, I feel one is slightly higher than the other. One can be documented in a consistent manner as:
```
debug_info_tuple {
  int64 byte_offset = 1;
  SourceRange source_range = 2;
  optional int64 source_range_tag = 3;
  // optional int64 foo = 4;
}
```
While the other might need more words to describe - when # of fields is 2 then use this protocol, when # of fields is 3 then use that protocol, when we add another field, then use yet another protocol...

It's our ad-hoc protocol anyway, I don't know whether it matters in the long run :-)

Reply:
I put them together because byte_offset and source_range_tag acted as identifier and source_range last as something that was being identified.

Reply:
Looking at the old deserialization code, I think this change is FC-breaking, because it will try to deserialize `tup_elems[1]` as a SourceRange.  But if you append source_tag_range to the end, then the old code will ignore it and you won't break FC.  I think that's enough of a reason to append, even if putting it in the middle is cleaner.

Reply:
Ah yes, thats for that catch.